crates/router/src/core/payments/helpers.rs

```diff
@@ -1920,6 +1936,73 @@ pub async fn create_customer_if_not_exist<'a, F: Clone, R, D>(
     ))
 }
 
+#[cfg(feature = "v1")]
+#[instrument(skip_all)]
+#[allow(clippy::type_complexity)]
+pub async fn get_customer_if_exists<'a, F: Clone, R, D>(
+    state: &SessionState,
+    operation: BoxedOperation<'a, F, R, D>,
+    payment_data: &mut PaymentData<F>,
+    req: Option<CustomerDetails>,
+    provider: &domain::Provider,
+) -> CustomResult<(BoxedOperation<'a, F, R, D>, Option<domain::Customer>), errors::StorageError> {
+    let merchant_id = provider.get_account().get_id();
+    let storage_scheme = provider.get_account().storage_scheme;
+    let key_store = provider.get_key_store();
+    let db = &*state.store;
+
+    // Same customer_id resolution as create_customer_if_not_exist
+    let customer_id = req
+        .as_ref()
+        .and_then(|r| r.customer_id.clone())
+        .or(payment_data.payment_intent.customer_id.clone());
+
+    let customer = match customer_id {
+        Some(customer_id) => {
+            let customer = db
+                .find_customer_optional_by_customer_id_merchant_id(
+                    &customer_id,
+                    merchant_id,
+                    key_store,
+                    storage_scheme,
+                )
+                .await?;
```

Comment:
customer_id is passed and if that is not available in the system then it should be a error, we should not fallback as guest here.